viewer/packages/cad-model/src/batching/MultiBufferBatchingManager.ts

```diff
@@ -0,0 +1,313 @@
+      this.removeSectorBatches(sectorId);
+    }
+
+    for (const { buffers } of this._instanceBatches.values()) {
+      buffers.forEach(batchBuffer => {
+        this._batchGroup.remove(batchBuffer.mesh);
+        batchBuffer.mesh.geometry.dispose();
+      });
+    }
+
+    this._sectorBatches.clear();
+    this._instanceBatches.clear();
+  }
+
+  private removeTreeIndicesFromMeshUserData(
+    mesh: THREE.Mesh,
+    updateRange: { byteOffset: number; byteCount: number },
+    instanceAttributes: { name: string; attribute: InterleavedBufferAttribute }[]
+  ): void {
+    const { byteOffset, byteCount } = updateRange;
+
+    const treeIndexAttribute = this.getTreeIndexAttribute(instanceAttributes);
+    const typedArray = treeIndexAttribute.data.array as TypedArray;
+    const instanceByteSize = treeIndexAttribute.data.stride * typedArray.BYTES_PER_ELEMENT;
+
+    const firstWholeInstanceIndex = byteOffset / instanceByteSize;
+    const firstInvalidInstanceIndex = (byteOffset + byteCount) / instanceByteSize;
+
+    for (let i = firstWholeInstanceIndex; i < firstInvalidInstanceIndex; i++) {
+      const treeIndex = treeIndexAttribute.getX(i);
+      decrementOrDeleteIndex(mesh.userData.treeIndices, treeIndex);
+    }
+  }
+
+  private processGeometries(parsedGeometry: Required<ParsedGeometry>, sectorBatch: SectorBatch) {
+    const instanceBatch = this.getOrCreateInstanceBatch(parsedGeometry);
+    this.batchInstanceAttributes(parsedGeometry.geometryBuffer, parsedGeometry.instanceId, instanceBatch, sectorBatch);
+  }
+
+  private batchInstanceAttributes(
+    bufferGeometry: BufferGeometry,
+    instanceId: string,
+    instanceBatch: InstanceBatch,
+    sectorBatch: SectorBatch
+  ) {
+    const interleavedBufferView = GeometryBufferUtils.getInstanceAttributesSharedView(bufferGeometry);
+    const batchBuffer = this.getBatchBufferToFill(instanceBatch);
+
+    const length = interleavedBufferView.byteLength;
+    const offset = interleavedBufferView.byteOffset;
+    const interleavedAttributesView = new Uint8Array(interleavedBufferView.buffer, offset, length);
+
+    const { batchId, bufferIsReallocated, updateRange } = batchBuffer.buffer.add(interleavedAttributesView);
+
+    const sectorInstanceData: SectorInstanceData = { batchBuffer, batchId, instanceCount: 0 };
+
+    sectorBatch.instanceBatches.set(instanceId, sectorInstanceData);
+
+    GeometryBufferUtils.getAttributes(batchBuffer.mesh.geometry, InterleavedBufferAttribute).forEach(
+      ({ attribute }) => {
+        attribute.data.needsUpdate = true;
+      }
+    );
+
+    const sourceInstanceAttributes = GeometryBufferUtils.getAttributes(bufferGeometry, InterleavedBufferAttribute);
+
+    const treeIndexInterleavedAttribute = this.getTreeIndexAttribute(sourceInstanceAttributes);
+
+    for (let i = 0; i < treeIndexInterleavedAttribute.count; i++) {
+      incrementOrInsertIndex(batchBuffer.mesh.userData.treeIndices, treeIndexInterleavedAttribute.getX(i));
+    }
```

Comment:
Could we have an `addTreeIndicesToMeshUserData` method, similar to the `removeTreeIndicesFromMeshUserData` method?